lib/poxa/web_hook/event_table.ex

```diff
@@ -0,0 +1,31 @@
+defmodule Poxa.WebHook.EventTable do
+  @table_name :web_hook_events
+
+  def new do
+    :ets.new(@table_name, [:bag, :public, :named_table])
+  end
+
+  def init do
+    case :ets.info(@table_name) do
+      :undefined -> new
+      information_list when information_list |> is_list -> @table_name
+    end
+  end
+
+  def insert(event, delay \\ 0) do
+    true = :ets.insert(@table_name, {time_ms + delay, event})
+  end
+
+  def all do
+    :ets.select(@table_name, [{{:"$1", :"$2"}, [], [:"$2"]}])
+  end
+
+  def ready(timestamp \\ time_ms + 1) do
+    {timestamp, :ets.select(@table_name, [{{:"$1", :"$2"}, [{:<, :"$1", timestamp}], [:"$2"]}]) }
+  end
+
+  def clear, do: :ets.delete_all_objects(@table_name)
```

Comment:
If clear is used just on tests I would rather just have this on the test itself calling directly `:ets`